From e5ad47fcf342211c83da9bfe1d11c3abf854d942 Mon Sep 17 00:00:00 2001
Message-Id: <e5ad47fcf342211c83da9bfe1d11c3abf854d942.1481738366.git.ymankad@redhat.com>
In-Reply-To: <be7cc66c94d4b6c77f9f1e674d4783c7d1c3efac.1481738366.git.ymankad@redhat.com>
References: <be7cc66c94d4b6c77f9f1e674d4783c7d1c3efac.1481738366.git.ymankad@redhat.com>
From: Vlad Yasevich <vyasevic@redhat.com>
Date: Tue, 13 Dec 2016 03:04:49 -0500
Subject: [CHANGE 10/10] net/rtl8139: update network information when macaddr
 is changed in guest
To: ymankad@redhat.com

RH-Author: Vlad Yasevich <vyasevic@redhat.com>
Message-id: <1481598289-10330-5-git-send-email-vyasevic@redhat.com>
Patchwork-id: 73043
O-Subject: [RHEL6.9 qemu-kvm PATCH 4/4] net/rtl8139: update network information when macaddr is changed in guest
Bugzilla: 1300626
RH-Acked-by: Vlad Yasevich <vyasevic@redhat.com>
RH-Acked-by: Stefan Hajnoczi <stefanha@redhat.com>
RH-Acked-by: Xiao Wang <jasowang@redhat.com>
RH-Acked-by: Laurent Vivier <lvivier@redhat.com>

From: Amos Kong <akong@redhat.com>

rtl8139 has same problem as e1000, nic info isn't updated when macaddr
is changed in guest.

This patch updates the nic info when the last bit of macaddr is written.

Signed-off-by: Amos Kong <akong@redhat.com>
Signed-off-by: Stefan Hajnoczi <stefanha@redhat.com>
(cherry picked from commit 23c37c37f0280761072c23bf67d3a4f3c0ff25aa)

Conflicts:
	hw/net/rtl8139.c
        - file location and lack of qemu_get_queue().

Signed-off-by: Vladislav Yasevich <vyasevic@redhat.com>
Signed-off-by: Yash Mankad <ymankad@redhat.com>
---
 hw/rtl8139.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/hw/rtl8139.c b/hw/rtl8139.c
index 12f829d..60c331d 100644
--- a/hw/rtl8139.c
+++ b/hw/rtl8139.c
@@ -2698,9 +2698,13 @@ static void rtl8139_io_writeb(void *opaque, uint8_t addr, uint32_t val)
 
     switch (addr)
     {
-        case MAC0 ... MAC0+5:
+        case MAC0 ... MAC0+4:
             s->phys[addr - MAC0] = val;
             break;
+        case MAC0+5:
+            s->phys[addr - MAC0] = val;
+            qemu_format_nic_info_str(&s->nic->nc, s->phys);
+            break;
         case MAC0+6 ... MAC0+7:
             /* reserved */
             break;
-- 
2.7.4

